math/gretl/Portfile

```diff
@@ -0,0 +1,119 @@
+                    user interface and a command-line interface.
+
+homepage            https://gretl.sourceforge.net/
+platforms           darwin
+
+master_sites        https://prdownloads.sourceforge.net/gretl/
+use_xz              yes
+
+checksums           rmd160  0df4dbb09b6ffb8d403d5b7fe67d8834a88b1e9b \
+                    sha256  ce28f7dfb3ff31e982c2f31a8766d4137a1dbf429ffca5ba7d49763a03338ee2 \
+                    size    13256260
+
+depends_build       port:pkgconfig \
+                    port:gmake
+
+depends_lib         port:fftw-3 \
+                    port:gettext \
+                    port:glib2 \
+                    port:json-glib \
+                    port:libxml2 \
+                    port:readline
+
+configure.args-append \
+                    --disable-avx \
+                    --disable-gmp \
+                    --disable-gui \
+                    --disable-openmp \
+                    --disable-www \
+                    --disable-xdg \
+                    --disable-xdg-utils \
+                    --without-gsf \
+                    --without-libR \
+                    --without-mpi \
+                    --without-odbc \
+                    --with-gmake \
+                    --with-readline-prefix=${prefix}
+
+compiler.c_standard 1999
+configure.cflags-append -std=c99
+
+build.cmd           gmake
+build.target        ""
+
+variant curl description {Enable downloads via libcurl} {
+    configure.args-replace --disable-www --enable-www
+    depends_lib-append \
+                    port:curl
+}
+
+variant gmp description {Enable multi-precision computation via GNU MP} {
+    configure.args-replace --disable-gmp --enable-gmp
+    configure.args-append \
+                    --with-gmp-prefix=${prefix} \
+                    --with-mpfr-prefix=${prefix}
+
+    depends_lib-append \
+                    port:gmp \
+                    port:mpfr
+}
+
+variant gui requires curl description {Build the Graphical User Interface} {
+    configure.args-replace --disable-gui --enable-gui
+    depends_lib-append \
+                    port:gdk-pixbuf2 \
+                    port:gtk3 \
+                    port:gtksourceview4
+
+    depends_run-append \
+                    port:adwaita-icon-theme \
+                    port:gnuplot
+
+    require_active_variants gnuplot pangocairo
+}
+
+variant odbc description {Enable ODBC data connections} {
+    configure.args-replace --without-odbc --with-odbc
+    depends_lib-append port:unixodbc
+}
+
+variant zip description {Support for compressed files via libgsf} {
+    configure.args-replace --without-gsf --with-gsf
+    depends_lib-append \
+                    port:libgsf
+}
+
+variant quartz requires gui {}
+
+if {[variant_isset quartz]} {
+    PortGroup               app 1.0
+    app.icon                xdg/64/gretl.png
+
+    require_active_variants gtk3 quartz
+
+    depends_lib-append \
+                    port:gtk-osx-application-gtk3
```

Comment:
It seems to me this whole code block can just be folded in the `quartz` variant declaration above, right?

Reply:
I was following the example of other Portfiles, which declared variants and then later used `variant_isset` to implement their logic. I'm not sure what the underlying rationale is – so I was just being superstitious.